mlflow/langchain/_langchain_autolog.py

```diff
@@ -105,53 +106,77 @@ def _update_langchain_model_config(model):
         return True
 
 
-def _inject_mlflow_callback(func_name, mlflow_callback, args, kwargs):
-    if func_name == "invoke":
-        from langchain.schema.runnable.config import RunnableConfig
+def _inject_callbacks(original_callbacks, new_callbacks):
+    """
+    Inject list of callbacks into the original callbacks.
+    For RunnableConfig, the callbacks is defined as List[BaseCallbackHandler] or BaseCallbackManager
+    https://github.com/langchain-ai/langchain/blob/ed980601e1c630f996aabf85df5cb26178e53099/libs/core/langchain_core/callbacks/base.py#L636
+    """
+    from langchain_core.callbacks.base import BaseCallbackManager
 
-        in_args = False
-        # `config` is the second positional argument of runnable.invoke function
-        # https://github.com/langchain-ai/langchain/blob/7d444724d7582386de347fb928619c2243bd0e55/libs/core/langchain_core/runnables/base.py#L468
-        if len(args) >= 2:
-            config = args[1]
-            in_args = True
-        else:
-            config = kwargs.get("config", None)
-        if config is None:
-            callbacks = [mlflow_callback]
-            config = RunnableConfig(callbacks=callbacks)
-        else:
-            callbacks = config.get("callbacks") or []
-            if not isinstance(callbacks, list):
-                callbacks = [callbacks]
-            config["callbacks"] = [*callbacks, mlflow_callback]
-        if in_args:
-            args = (args[0], config) + args[2:]
+    if isinstance(original_callbacks, BaseCallbackManager):
+        for callback in new_callbacks:
+            original_callbacks.add_handler(callback)
+        return original_callbacks
+    if not isinstance(original_callbacks, list):
+        original_callbacks = [original_callbacks]
+    original_callbacks.extend(new_callbacks)
+    return original_callbacks
+
+
+def _inject_callbacks_for_runnable(mlflow_callbacks, args, kwargs):
+    """
+    `config` is the second positional argument of runnable.invoke and runnable.batch functions
+    https://github.com/langchain-ai/langchain/blob/7d444724d7582386de347fb928619c2243bd0e55/libs/core/langchain_core/runnables/base.py#L468
+    https://github.com/langchain-ai/langchain/blob/ed980601e1c630f996aabf85df5cb26178e53099/libs/core/langchain_core/runnables/base.py#L600-L607
+
+    """
+    from langchain.schema.runnable.config import RunnableConfig
+
+    in_args = False
+    if len(args) >= 2:
+        config = args[1]
+        in_args = True
+    else:
+        config = kwargs.get("config")
+    if config is None:
+        config = RunnableConfig(callbacks=mlflow_callbacks)
```

Comment:
```suggestion
        config = kwargs.get("config", RunnableConfig(callbacks=mlflow_callbacks))
```

Reply:
Emm I want to avoid situations when kwargs["config"]=None, also we can't combine them into one line because if config==kwargs.get("config") we'll inject mlflow_callbacks into it.

Reply:
Ah that makes sense. I'd use `or` here but it's up to you:

```suggestion
        config = kwargs.get("config")
    
    config = config or RunnableConfig(callbacks=mlflow_callbacks)
```

Reply:
Got it, makes sense!